community/index/src/main/java/org/neo4j/index/internal/gbptree/GBPTree.java

```diff
@@ -754,16 +772,25 @@ public void close( Consumer<PageCursor> headerWriter ) throws IOException
     private void close( Header.Writer headerWriter ) throws IOException
     {
         writer.close();
-
+        boolean didCheckpoint = false;
+        boolean anyWrites = false;
         try
         {
-            // Perform a checkpoint before closing. If no changes has happened since last checkpoint,
-            // no new checkpoint will be created.
-            checkpoint( IOLimiter.unlimited(), headerWriter );
+            if ( health.isHealthy() )
+            {
+                // Perform a checkpoint before closing. If no changes has happened since last checkpoint,
+                // no new checkpoint will be created.
+                anyWrites = checkpoint( IOLimiter.unlimited(), headerWriter );
+                didCheckpoint = true;
+            }
         }
         finally
         {
             pagedFile.close();
+            String checkpointMessage = didCheckpoint ?
```

Comment:
I think will keep it since the problem we are addressing is when we do checkpoints even though we shouldn't.